donuts-revisited.js

```diff
@@ -0,0 +1,74 @@
+/*
+ * Programming Quiz: Donuts Revisited (7-6)
+ */
+
+/*
+ * QUIZ REQUIREMENTS
+ * - Your code sshould have an array named `donuts`
+ * - Your `donuts` array should call the `forEach()` method
+ * - Your `forEach()` loop should output the donut summaries
+ * - BE CAREFUL ABOUT THE PUNCTUATION, SPACES, AND EXACT WORDS TO BE PRINTED.
+
+Use the forEach() method 
+to loop over the array 
+and print out the following donut summaries using console.log()
+
+Jelly donuts cost $1.22 each
+Chocolate donuts cost $2.45 each
+Cider donuts cost $1.59 each
+Boston Cream donuts cost $5.99 each
+
+
+ */
+
+// This is an ARRAY of OBJECTS. 
+var donuts = [
+    { type: "Jelly", cost: 1.22 },
+    { type: "Chocolate", cost: 2.45 },
+    { type: "Cider", cost: 1.59 },
+    { type: "Boston Cream", cost: 5.99 }
+];
+
+// your code goes here
+donuts.forEach(printFunction);
+function printFunction(item, index, array){
+	console.log(donuts[index].type + " donuts cost $" + donuts[index].cost + " each");
+}
+
+
+
+
+// Udacity answer
+
+/*
+
+donuts.forEach(function(donut){
+
+    // donut represents a single element of donuts array
+    // donut is an object, therefore you can access its properties using a dot notation
+    console.log(donut.type+" donuts cost $"+donut.cost+" each");
+});
+*/
+
+
```

Comment:
Cual metodo de forEach se usa mas seguido o es mas comun?
@andres-ngm 

Tratando de resolver y volver a mirar el forEach, encontre 3 diferentes maneras:
- Uno de ejemplo en Udacity
- Uno en Mozilla
- Uno en W3schools

Reply:
Deja me ver si estoy entendiendo, como tal solo existe un metodo forEach, pero me imagino que te refieres a las diferentes formas que hay para iterar arreglos.

Hoy en dia para iterar un arreglo lo mas comun es usar el metodo `forEach`.
`
myarray.forEach(function(item, index, array) {
 console.log(item);
});
`

Este es el for conventional, la ventaja es que uno puede manipular el cursor. Imagina que quieres iterar el arreglo a una velocidad del doble (brincando el cursor de 2 en 2).
`
for (let index = 0; index < myarray.length; index+=2) {
    console.log(myarray[index]);
}
`

Este es una mejora el for conventional, podriamos decir que es el homologo del de python, si solo te interesa iterar el arreglo sin importar el indice esta seria la opcion.
`
for (const element of myarray) {
   console.log(element);
}
`